### PR TITLE
Allow for downloading of unlabeled prediction data

### PR DESCRIPTION
**Pull request by acrellin**
- Fixes bug resulting in error if attempt to download unlabeled prediction results was made.
- Adds corresponding test.
- Updates test fixtures.

### CODE REVIEW

File: cesium_app/tests/fixtures.py [hunk 1]
```diff
@@ -150,18 +149,20 @@ def create_test_model(fset, model_type='RandomForestClassifier'):
...
-        The dataset on which prediction will be performed.
+        Dummy dataset used to create prediction instance.
+    featureset : `models.Featureset` instance
+        The featureset on which prediction will be performed.
     model : `models.Model` instance
         The model to use to create prediction.
 
     """
-    fset, data = featurize.load_featureset(model.featureset.file.uri)
+    fset, data = featurize.load_featureset(featureset.file.uri)
```
Comment (bnaul) [inline cesium_app/tests/fixtures.py]: Why is this change needed? Aren't the two equivalent?

File: cesium_app/tests/frontend/test_predict.py [hunk 2]
```diff
@@ -204,12 +204,30 @@ def test_download_prediction_csv_class(driver):
             os.remove('/tmp/cesium_prediction_results.csv')
 
 
+def test_download_prediction_csv_class_unlabeled(driver):
```
Comment (bnaul) [inline cesium_app/tests/frontend/test_predict.py]: `def test_download_prediction_csv_class_unlabeled(driver, tmpdir):`

File: cesium_app/tests/frontend/test_predict.py [hunk 3]
```diff
@@ -204,12 +204,30 @@ def test_download_prediction_csv_class(driver):
...
+    driver.get('/')
+    with create_test_project() as p, create_test_dataset(p, label_type=None) as ds,\
+         create_test_featureset(p) as fs,\
+         create_test_featureset(p, label_type=None) as unlabeled_fs,\
+         create_test_model(fs, model_type='LinearSGDClassifier') as m,\
+         create_test_prediction(ds, unlabeled_fs, m):
+        _click_download(p.id, driver)
```
Comment (bnaul) [inline cesium_app/tests/frontend/test_predict.py]: If we can change `_click_download` to use the `tmpdir` fixture that `pytest` provides then we don't have to handle the `finally: <remove file>` part

File: cesium_app/tests/fixtures.py (hunk 1 above)
Comment (acrellin) [reply cesium_app/tests/fixtures.py]: No, that's the whole point of this change - to test w/ unlabeled data.

Comment (acrellin) [reply cesium_app/tests/fixtures.py]: See the new test - there are two feature sets created there - one labeled, used to build the model, the other unlabeled, used in the prediction.

File: cesium_app/tests/frontend/test_predict.py (hunk 3 above)
Comment (acrellin) [reply cesium_app/tests/frontend/test_predict.py]: Sure, that's a good idea.

**Review by bnaul — APPROVED** [review]
